### PR TITLE
Reinstate "Update RareClusterStateIT to work with ... (#87922)"

### PR DESCRIPTION
**Pull request by DaveCTurner**
Reverts #90720 and adds some additional logging to further investigate the failure here.

Relates #87922

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/cluster/coordination/RareClusterStateIT.java [hunk 1]
```diff
@@ -371,6 +408,12 @@ public void testDelayedMappingPropagationOnReplica() throws Exception {
...
         // and not just because it takes time to replicate the indexing request to the replica
         Thread.sleep(100);
+        final var results = "["
+            + (putMappingResponse.isDone() ? Strings.toString(putMappingResponse.get()) : "null")
+            + ","
+            + (docIndexResponse.isDone() ? Strings.toString(docIndexResponse.get()) : "null")
+            + "]";
+        assertEquals("[null,null]", results);
         assertFalse(putMappingResponse.isDone());
         assertFalse(docIndexResponse.isDone());
```
Comment (idegtiarenko) [inline server/src/internalClusterTest/java/org/elasticsearch/cluster/coordination/RareClusterStateIT.java:418]: It looks like there is no way with hamcrest to combine multiple assertions on different objects into one.
Looks like another reason to use assertj: https://stackoverflow.com/questions/47397525/multiply-conditions-set-in-assertj-assertions

**Review by idegtiarenko — APPROVED** [review]

**Comment by pxsalehi** [comment]
This seems to still fail (at least for me): https://gradle-enterprise.elastic.co/s/hs6e6lwwd6kou

**Comment by DaveCTurner** [comment]
How about now (after dc638d97de33bb3f0ad8af08a72d573d1fecf3b3) @pxsalehi?

IMO the failing assertion is kinda bogus anyway, it doesn't really matter if the put-mapping request has completed or not as long as the indexing request is still outstanding.

**Comment by pxsalehi** [comment]
> How about now (after [dc638d9](https://github.com/elastic/elasticsearch/commit/dc638d97de33bb3f0ad8af08a72d573d1fecf3b3)) @pxsalehi?

Not anymore! 😃 🎉  
https://gradle-enterprise.elastic.co/s/lhvvwt4xtntdq

**Comment by DaveCTurner** [comment]
Great, thanks for checking.

**Pull request merged**